### PR TITLE
initial tax service

### PR DESCRIPTION
**Pull request by nk-coding**
https://frontend.gropius.duckdns.org/components/30a21215-1734-4622-a937-0cee6e4a7379/issues/32b1c609-b482-4885-b34f-beb41092d8cc

- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

### CODE REVIEW

File: src/main/kotlin/org/misarch/tax/graphql/model/TaxRateVersion.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    val rate: Double,
+    @property:GraphQLDescription("Version of the TaxRateVersion")
+    val version: Int,
+    @property:GraphQLDescription("Time stamp when the TaxRateVersion was created")
+    val createdAt: OffsetDateTime,
+    private val taxRateId: UUID
+) : Node(id) {
+
+    @GraphQLDescription("The associated TaxRate.")
+    fun currentVersion(
```
Comment (delvh) [inline src/main/kotlin/org/misarch/tax/graphql/model/TaxRateVersion.kt]: Is that the correct description?
Doesn't sound like what this method is supposed to do…

File: src/main/kotlin/org/misarch/tax/graphql/model/connection/TaxRateVersionConnection.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * A GraphQL connection for [TaxRateVersion]s.
+ *
+ * @param first The maximum number of items to return
+ * @param skip The number of items to skip
+ * @param predicate The predicate to filter the items by
+ * @param order The order to sort the items by
+ * @param repository The repository to fetch the items from
+ * @param applyJoin A function to apply a join to the query
+ */
+@GraphQLDescription("A connection to a list of `User` values.")
```
Comment (delvh) [inline src/main/kotlin/org/misarch/tax/graphql/model/connection/TaxRateVersionConnection.kt]:
```suggestion
@GraphQLDescription("A connection to a list of `TaxRate` values.")

File: src/main/kotlin/org/misarch/tax/graphql/model/TaxRateVersion.kt (hunk 1 above)
Comment (nk-coding) [reply src/main/kotlin/org/misarch/tax/graphql/model/TaxRateVersion.kt]: fixed with 822bc48b77cd7baadb1ed3080d6cb95e481f47bd

File: src/main/kotlin/org/misarch/tax/graphql/model/connection/TaxRateVersionConnection.kt (hunk 2 above)
Comment (nk-coding) [reply src/main/kotlin/org/misarch/tax/graphql/model/connection/TaxRateVersionConnection.kt]: fixed with 822bc48b77cd7baadb1ed3080d6cb95e481f47bd

**Review by delvh — APPROVED** [review]

**Pull request merged**